include/seqan3/alignment/scoring/detail/simd_matrix_scoring_scheme.hpp

```diff
@@ -0,0 +1,190 @@
+     * \param[in] scoring_scheme The scoring scheme to initialise the vectorised match and mismatch score with.
+     */
+    constexpr void initialise_from_scalar_scoring_scheme(scoring_scheme_t const & scoring_scheme)
+    {
+        using score_t = decltype(std::declval<scoring_scheme_t const &>().score(alphabet_t{}, alphabet_t{}));
+        using simd_scalar_t = typename simd_traits<simd_score_t>::scalar_type;
+
+        // Check if the scoring scheme match and mismatch scores do not overflow with the respective scalar type.
+        if constexpr (sizeof(simd_scalar_t) < sizeof(score_t))
+        {
+            if (min_or_max_exceeded<score_t, simd_scalar_t>(scoring_scheme))
+                throw std::invalid_argument{"The selected scoring scheme score overflows "
+                                            "for the selected scalar type of the simd type."};
+        }
+
+        internal_scoring_scheme = scoring_scheme;
+    }
+
+    /*!\brief Check if any score in the scoring scheme matrix exceeds the min or max value allowed by the simd vector.
+     * \param[in] scoring_scheme The scoring scheme to check for min or max violations.
+     * \tparam score_t The type of the scores in the scoring scheme matrix.
+     * \tparam simd_scalar_t The type of the simd vector.
+     * \returns Returns true if it encounters a value larger than the maximum or lower than the minimum allowed by
+     *          the simd vector, and false if otherwise.
+     */
+    template <typename score_t, typename simd_scalar_t>
+    constexpr bool min_or_max_exceeded(scoring_scheme_t const & scoring_scheme)
+    {
+        score_t max_val = static_cast<score_t>(std::numeric_limits<simd_scalar_t>::max());
+        score_t min_val = static_cast<score_t>(std::numeric_limits<simd_scalar_t>::lowest());
+
+        for (uint8_t i = 0; i < alphabet_size<alphabet_t>; ++i)
+        {
+            for (uint8_t j = 0; j < alphabet_size<alphabet_t>; ++j)
+            {
+                score_t tmp_score = scoring_scheme.score(assign_rank_to(i, alphabet_t{}),
+                                                         assign_rank_to(j, alphabet_t{}));
+
+                if (tmp_score > max_val || tmp_score < min_val)
+                    return true;
+            }
+        }
+
+        return false;
+    }
+
+    /*!\brief Checks whether the given value is a padded value or not.
+     * \param[in] value The value to check if it is padded.
+     * \returns Returns true if the value is padded and false otherwise.
+     *
+     * \details This check is done by casting the scalar type of the value into an unsigned value, and then checking
+     *          to see if this is larger than the alphabet size, which would imply that it is a padded value.
+     */
+    constexpr bool is_padded(typename simd_traits<simd_score_t>::scalar_type const & value) const
+    {
+        typedef typename std::make_unsigned<typename simd_traits<simd_score_t>::scalar_type>::type unsigned_scalar_t;
```

Comment:
please use the `using
```suggestion
        using unsigned_scalar_t = typename std::make_unsigned<typename simd_traits<simd_score_t>::scalar_type>::type;
```

Reply:
Can I ask what the difference is between typedef and using? Is it just for consistency's sake?

Reply:
`typedef` was used before C++11 and `using` is now the more appropriate way. It is in fact an assignment of a type and thus it is easier to read. Other from that there is not much of a difference.